spicy/toolchain/src/compiler/parser/parser.yy

```diff
@@ -383,6 +385,16 @@ hook_decl     : ON unit_hook_id unit_hook        { ID unit = $2.namespace_();
                                                  }
               ;
 
+/* field_decl follows a similar set of rules as unit_field itself, see below. */
```

Comment:
It would be great if we could move the RHS-parts of `field_decl` and `unit_field` into its own rule to avoid duplication.

Reply:
they have different structure, don't think that would work without making it more messy